sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2.scala

```diff
@@ -62,7 +62,7 @@ case class WriteToDataSourceV2Exec(writer: DataSourceV2Writer, query: SparkPlan)
 
     try {
       val runTask = writer match {
-        case w: ContinuousWriter =>
+        case w: StreamWriter =>
```

Comment:
Will we hit this branch under micro batch mode?

Reply:
No - the writer will be wrapped in a MicroBatchWriter which doesn't match StreamWriter. I agree this is a bit unclear and would welcome suggestions on how to refactor.

Reply:
I don't have a better idea, but at least we should add some comments here to explain this.